Reading secret from Vault doesn't work

Given a credential configured with a Vault path and a Claim for a service, if we try to bind an application to the service using this claim, an expection is thrown: 

```
HTTP Request to /applications/claim/2 failed, error id: ea46dd0d-5e2b-4f10-b108-191bca53bf8c-1: org.jboss.resteasy.spi.UnhandledException: io.quarkus.vault.runtime.client.VaultClientException code=403 body={"errors":["1 error occurred:\n\t* permission denied\n\n"]}

        at org.jboss.resteasy.core.ExceptionHandler.handleApplicationException(ExceptionHandler.java:105)
        at org.jboss.resteasy.core.ExceptionHandler.handleException(ExceptionHandler.java:359)
        at org.jboss.resteasy.core.SynchronousDispatcher.writeException(SynchronousDispatcher.java:218)
        at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:519)
        at org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:261)
        at org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:161)
        at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
        at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:164)
        at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:247)
        at io.quarkus.resteasy.runtime.standalone.RequestDispatcher.service(RequestDispatcher.java:82)
        at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.dispatch(VertxRequestHandler.java:147)
        at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler$1.run(VertxRequestHandler.java:93)
        at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:576)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
        at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
        at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: io.quarkus.vault.runtime.client.VaultClientException code=403 body={"errors":["1 error occurred:\n\t* permission denied\n\n"]}

        at io.quarkus.vault.runtime.client.VertxVaultClient.throwVaultException(VertxVaultClient.java:206)
        at io.quarkus.vault.runtime.client.VertxVaultClient.lambda$exec$1(VertxVaultClient.java:155)
        at io.smallrye.mutiny.unchecked.UncheckedFunction.lambda$toFunction$0(UncheckedFunction.java:45)
        at io.smallrye.context.impl.wrappers.SlowContextualFunction.apply(SlowContextualFunction.java:21)
        at io.smallrye.mutiny.operators.uni.UniOnItemTransform$UniOnItemTransformProcessor.onItem(UniOnItemTransform.java:36)
        at io.smallrye.mutiny.operators.uni.UniFailOnTimeout$UniFailOnTimeoutProcessor.onItem(UniFailOnTimeout.java:69)
        at io.smallrye.mutiny.vertx.AsyncResultUni.lambda$subscribe$1(AsyncResultUni.java:35)
        at io.smallrye.mutiny.vertx.DelegatingHandler.handle(DelegatingHandler.java:25)
        at io.vertx.ext.web.client.impl.HttpContext.handleDispatchResponse(HttpContext.java:397)
        at io.vertx.ext.web.client.impl.HttpContext.execute(HttpContext.java:384)
        at io.vertx.ext.web.client.impl.HttpContext.next(HttpContext.java:362)
        at io.vertx.ext.web.client.impl.HttpContext.fire(HttpContext.java:329)
        at io.vertx.ext.web.client.impl.HttpContext.dispatchResponse(HttpContext.java:291)
        at io.vertx.ext.web.client.impl.HttpContext.lambda$null$7(HttpContext.java:507)
        at io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:264)
        at io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:246)
        at io.vertx.core.impl.EventLoopContext.lambda$runOnContext$0(EventLoopContext.java:43)
        at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174)
        at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167)
        at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:566)
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
        ... 2 more
```

--- a/app/src/main/java/io/halkyon/resource/GreetingResource.java
+++ b/app/src/main/java/io/halkyon/resource/GreetingResource.java
@@ -11,9 +11,6 @@ import io.quarkus.vault.VaultKVSecretEngine;
 
 @Path("/hello")
 public class GreetingResource {
-    //
-    // @ConfigProperty(name = "a-private-key")
-    // String privateKey;
 
     @Inject
     VaultKVSecretEngine kvSecretEngine;
@@ -22,13 +19,7 @@ public class GreetingResource {
     @Path("/secrets/{vault-path}")
     @Produces(MediaType.TEXT_PLAIN)
     public String getSecrets(@PathParam("vault-path") String vaultPath) {
-        return kvSecretEngine.readSecret("myapps/vault-quickstart/" + vaultPath).toString();
+        return kvSecretEngine.readSecret("primaza/" + vaultPath).toString();
     }
 
-    // @GET
-    // @Path("/private-key")
-    // @Produces(MediaType.TEXT_PLAIN)
-    // public String privateKey() {
-    // return privateKey;
-    // }
 }

--- a/app/src/main/resources/application.properties
+++ b/app/src/main/resources/application.properties
@@ -16,6 +16,5 @@ quarkus.hibernate-orm.database.generation=drop-and-create
 quarkus.vault.authentication.userpass.username=bob
 quarkus.vault.authentication.userpass.password=sinclair
 
-%prod.quarkus.vault.kv-secret-engine-version=1
-%prod.quarkus.vault.kv-secret-engine-mount-path=kv
+%prod.quarkus.vault.kv-secret-engine-mount-path=kv2
 

--- a/app/src/main/resources/application.properties
+++ b/app/src/main/resources/application.properties
@@ -9,11 +9,13 @@ quarkus.datasource.db-kind=h2
 quarkus.hibernate-orm.database.generation=drop-and-create
 
 # vault url
-%prod.quarkus.vault.url=http://localhost:8200
+%prod.quarkus.vault.url=http://vault.65.108.212.158.nip.io
+%dev.quarkus.vault.url=http://vault.127.0.0.1.nip.io
 
 # vault authentication
-%prod.quarkus.vault.authentication.userpass.username=bob
-%prod.quarkus.vault.authentication.userpass.password=sinclair
+quarkus.vault.authentication.userpass.username=bob
+quarkus.vault.authentication.userpass.password=sinclair
+quarkus.log.category."io.quarkus.vault".level=DEBUG
+quarkus.vault.kv-secret-engine-version=1
+quarkus.vault.kv-secret-engine-mount-path=kv
 
-# path within the kv secret engine where is located the vault-quickstart secret configuration
-%prod.quarkus.vault.secret-config-kv-path=myapps/vault-quickstart/config

--- a/app/src/main/resources/application.properties
+++ b/app/src/main/resources/application.properties
@@ -15,7 +15,7 @@ quarkus.hibernate-orm.database.generation=drop-and-create
 # vault authentication
 quarkus.vault.authentication.userpass.username=bob
 quarkus.vault.authentication.userpass.password=sinclair
-quarkus.log.category."io.quarkus.vault".level=DEBUG
-quarkus.vault.kv-secret-engine-version=1
-quarkus.vault.kv-secret-engine-mount-path=kv
+
+%prod.quarkus.vault.kv-secret-engine-version=1
+%prod.quarkus.vault.kv-secret-engine-mount-path=kv
 

--- a/scripts/vault.sh
+++ b/scripts/vault.sh
@@ -180,7 +180,7 @@ function registerUser() {
   fi
 
   log BLUE "Creating a new vault user: ${VAULT_USER}, password: ${VAULT_PASSWORD} having as policy: ${VAULT_POLICY_NAME}"
-  #vaultExec "vault write auth/userpass/users/${VAULT_USER} password=${VAULT_PASSWORD} policies=${VAULT_POLICY_NAME}"
+  vaultExec "vault write auth/userpass/users/${VAULT_USER} password=${VAULT_PASSWORD} policies=${VAULT_POLICY_NAME}"
 }
 
 case $1 in
@@ -210,7 +210,6 @@ createTokensKubernetesSecret
 createUserPolicy
 registerUser
 loginAsUser
-vaultExec "vault kv put kv/primaza/hello target=world"
 
 log YELLOW "Temporary folder containing created files: ${TMP_DIR}"
 log YELLOW "Vault Root Token: $(jq -r ".root_token" ${TMP_DIR}/cluster-keys.json)"

